### ISSUE TITLE
AWS::StepFunctions::StateMachine - (JSONata) false-positive finding `E3601` for `Output` in `Succeed` state

### ISSUE THREAD

**Issue by r-heimann** [aws-cloudformation/cfn-lint#4297]
### CloudFormation Lint Version

cfn-lint 1.41.0

### What operating system are you using?

Windows 11

### Describe the bug

cfn-lint incorrectly shows the error `[cfn-lint] E3601: Additional properties are not allowed ('Output' was unexpected)` when using `Output` in a `Succeed` state:

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: Test

Resources:
  StepFunctionRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Effect: Allow
            Principal:
              Service: states.amazonaws.com
            Action: sts:AssumeRole

  StateMachine:
    Type: AWS::StepFunctions::StateMachine
    Properties:
      RoleArn: !GetAtt StepFunctionRole.Arn
      Definition:
        QueryLanguage: JSONata
        StartAt: Pass
        States:
          Pass:
            Type: Pass
            Next: Succeed
          Succeed:
            Type: Succeed
            Output: {} # <------------------------------
```

### Expected behavior

cfn-lint doesn't give `E3601` when using `Output` in a `Succeed` state.

### Reproduction template

See above.

**Comment by mrrip40danyysamphwp** [comment]
This should be the fix.
[a5dea79d4305.zip](https://github.com/user-attachments/files/23651844/a5dea79d4305.zip)
